### PR TITLE
StrippableSystem doafter overhaul

### PR DESCRIPTION
**Pull request by Krunklehorn**
## About the PR

Refactors Strippable DoAfter events to make them synchronous and organized.

### Strippable System & Component
- Synchronous DoAfters
- Made use of `TimeSpan`, `GetStripTimeModifiers()` and `ByRefEvent`
- Reorganized checks, removed some redundant ones
- Resolve pattern where useful
- Added more asserts
- Lots of cleanup

The DoAfters were grouped under one event to avoid copy-pasting eight separate cancel checks, asserts and function signatures.

Let me know if this is bad for performance and I'll roll them out instead.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### TimeSpans
`ThievingComponent`, `InventoryTemplatePrototype` and `ToggleableClothingSystem` use `TimeSpan` in places where they intersect with `StrippableComponent`.

**Changelog**

N/A

### CODE REVIEW

File: Content.Server/Strip/StrippableSystem.cs [hunk 1]
```diff
@@ -123,352 +120,524 @@ public override void StartOpeningStripper(EntityUid user, Entity<StrippableCompo
...
+        ///     Checks whether the item is in a user's active hand and whether it can be inserted into the inventory slot.
         /// </summary>
-        private async void PlaceActiveHandItemInInventory(
+        private bool CanStripInsertInventory(
             EntityUid user,
             EntityUid target,
             EntityUid held,
             string slot,
-            StrippableComponent component)
+            HandsComponent userHands)
```
Comment (metalgearsloth) [inline Content.Server/Strip/StrippableSystem.cs:199]: Make this Entity<HandsComponent?> instead and resolve it.

File: Content.Server/Strip/StrippableSystem.cs [hunk 2]
```diff
@@ -123,352 +120,524 @@ public override void StartOpeningStripper(EntityUid user, Entity<StrippableCompo
...
+        ///     Checks whether the item in the user's active hand can be inserted into one of the target's hands.
+        /// </summary>
+        private bool CanStripInsertHand(
+            EntityUid user,
+            EntityUid target,
+            EntityUid held,
+            string handName,
+            HandsComponent userHands,
+            HandsComponent? targetHands = null)
         {
```
Comment (metalgearsloth) [inline Content.Server/Strip/StrippableSystem.cs:403]: Same here.

File: Content.Server/Strip/StrippableSystem.cs [hunk 3]
```diff
@@ -123,352 +120,524 @@ public override void StartOpeningStripper(EntityUid user, Entity<StrippableCompo
...
+        /// <summary>
+        ///     Inserts the item in the user's active hand into the inventory slot.
+        /// </summary>
+        private void StripInsertInventory(
+            EntityUid user,
+            EntityUid target,
+            EntityUid held,
+            string slot,
+            HandsComponent? userHands = null)
+        {
```
Comment (metalgearsloth) [inline Content.Server/Strip/StrippableSystem.cs:284]: Here

File: Content.Server/Strip/StrippableSystem.cs [hunk 4]
```diff
@@ -123,352 +120,524 @@ public override void StartOpeningStripper(EntityUid user, Entity<StrippableCompo
...
+
+        /// <summary>
+        ///     Begins a DoAfter to insert the item in the user's active hand into the inventory slot.
+        /// </summary>
+        private void StartStripInsertInventory(
+            EntityUid user,
+            EntityUid target,
+            EntityUid held,
+            string slot,
+            HandsComponent? userHands = null)
```
Comment (metalgearsloth) [inline Content.Server/Strip/StrippableSystem.cs:239]: Here

File: Content.Server/Strip/StrippableSystem.cs [hunk 5]
```diff
@@ -123,352 +120,524 @@ public override void StartOpeningStripper(EntityUid user, Entity<StrippableCompo
...
-            _handsSystem.TryDrop(target, item, checkActionBlocker: false, handsComp: hands);
-            _handsSystem.PickupOrDrop(user, item, animateUser: !ev.Stealth, animate: !ev.Stealth, handsComp: userHands);
-            // hand update will trigger strippable update
-            _adminLogger.Add(LogType.Stripping, LogImpact.Medium,
-                $"{ToPrettyString(user):actor} has stripped the item {ToPrettyString(item):item} from {ToPrettyString(target):target}'s hands");
+            Debug.Assert(entity.Owner == ev.User);
+            Debug.Assert(ev.Target != null);
+            Debug.Assert(ev.Used != null);
+            Debug.Assert(ev.SlotOrHandName != null);
+
```
Comment (metalgearsloth) [inline Content.Server/Strip/StrippableSystem.cs:628]: Use DebugTools

**Review by metalgearsloth — APPROVED** [review]
Mainly just the Entity<T> stuff instead of passing components in manually.

**Comment by Krunklehorn** [comment]
Tests failed to build due to yesterday's bug.

**Comment by metalgearsloth** [comment]
I could update to master locally or I could see what happens.

**Pull request merged**